Restack X-Windows after being mapped to ensure EnterNotify events fire

medranocalvo opened the pull request:
```
* exwm-manage.el (exwm-manage--on-MapNotify, exwm-manage--init):
Restack X windows after being mapped in order to ensure they receive
an EnterNotify event when running under XQuartz.
```

This change fixes #422 for me.  Let me know of any comments.

ch11ng commented on exwm-manage.el:
```diff
@@ -665,6 +665,20 @@ border-width: %d; sibling: #x%x; stack-mode: %d"
       (exwm-manage--unmanage-window window t))))
 
+(defun exwm-manage--on-MapNotify (data _synthetic)
+  (let ((obj (make-instance 'xcb:UnmapNotify)))
```
You actually mean `xcb:MapNotify`.

ch11ng commented on exwm-manage.el:
```diff
@@ -665,6 +665,20 @@ border-width: %d; sibling: #x%x; stack-mode: %d"
+          (make-instance 'xcb:ConfigureWindow
+                         :window window
+                         :value-mask xcb:ConfigWindow:StackMode
+                         :stack-mode xcb:StackMode:Above))
```
We may receive MapNotify from other sources.  So perhaps we should test whether the X window is in `exwm--id-buffer-alist`?

ch11ng commented on exwm-manage.el:
```diff
@@ -692,6 +706,8 @@ border-width: %d; sibling: #x%x; stack-mode: %d"
               #'exwm-manage--on-ConfigureRequest)
   (xcb:+event exwm--connection 'xcb:MapRequest #'exwm-manage--on-MapRequest)
   (xcb:+event exwm--connection 'xcb:UnmapNotify #'exwm-manage--on-UnmapNotify)
+  (when (eq system-type 'darwin)
```
Please consider applying this 'workaround' to all platforms if it has no side-effect.  It would help reducing the testing scenarios.

medranocalvo commented:
> You actually mean xcb:MapNotify.

Corrected.

> We may receive MapNotify from other sources. So perhaps we should test whether the X window is in exwm--id-buffer-alist?

Corrected.

> Please consider applying this 'workaround' to all platforms if it has no side-effect. It would help reducing the testing scenarios.

It now applies to all platforms.  I can't foresee any bad effect except the useless restacking.  Please, test it in your environment.

Thank you for the review.
